src/main/java/org/quantumbadger/redreader/common/PrefsUtility.java

```diff
@@ -865,14 +867,52 @@ public static boolean network_tor(final Context context, final SharedPreferences
 		return result;
 	}
 
-	public static EnumSet<OptionsMenuUtility.OptionsMenuItemsPref> pref_menus_optionsmenu_items(final Context context, final SharedPreferences sharedPreferences) {
+	private static class AppbarItemInfo {
+		final Enum<OptionsMenuUtility.AppbarItemsPref> itemPref;
+		final int stringRes, defaultValue;
 
-		final Set<String> strings = getStringSet(R.string.pref_menus_optionsmenu_items_key, R.array.pref_menus_optionsmenu_items_items_default, context, sharedPreferences);
+		AppbarItemInfo(final Enum<OptionsMenuUtility.AppbarItemsPref> itemPref, final int stringRes, final int defaultValue) {
+			this.itemPref = itemPref;
+			this.stringRes = stringRes;
+			this.defaultValue = defaultValue;
+		}
+	}
 
-		final EnumSet<OptionsMenuUtility.OptionsMenuItemsPref> result = EnumSet.noneOf(OptionsMenuUtility.OptionsMenuItemsPref.class);
-		for(String s : strings) result.add(OptionsMenuUtility.OptionsMenuItemsPref.valueOf(General.asciiUppercase(s)));
+	public static Map<Enum<OptionsMenuUtility.AppbarItemsPref>, Integer> pref_menus_appbar_items(final Context context, final SharedPreferences sharedPreferences) {
```

Comment:
As before, this can be `EnumMap<OptionsMenuUtility.AppbarItemsPref, Integer>` for improved performance/space usage.